File: scripts/refresh-rankings.mjs
#!/usr/bin/env node
/**
 * Fast refresh: rankings only.
 *
 * Rankings are published weekly, so a light refresh needs just two API calls
 * plus the derived aggregates.  Everything else (biographies, match logs,
 * calendar) changes slowly and is handled by the full build.
 */
import { spawn } from 'node:child_process';
import { resolve } from 'node:path';
import { ROOT } from './lib.mjs';

// Chinese names are keyed by WTA player id, so a ranking refresh needs a new
// zh snapshot too or newly ranked players would have no Chinese name.
// Rankings drive the dashboard payload and the Chinese name map, so a light
// refresh rebuilds all three and re-verifies before anything is published.
const STEPS = [
  'fetch-rankings.mjs',
  'derive.mjs',
  'fetch-events.mjs',
  'fetch-zh.mjs',
  'generate-data.mjs',
  'build-site.mjs',
  'verify-dashboard.mjs',
];

for (const file of STEPS) {
  console.log(`\n▶ ${file}`);
  const code = await new Promise((res) =>
    spawn(process.execPath, [resolve(ROOT, 'scripts', file)], {
      stdio: 'inherit',
      cwd: ROOT,
      env: process.env,
    }).on('close', res),
  );
  if (code !== 0) process.exit(code ?? 1);
}

console.log('\n✓ Rankings refreshed.');

File: pyscripts/refresh_rankings.py
#!/usr/bin/env python3
"""
Light weekly refresh: rankings only.

Rankings are published on Mondays, so a refresh needs the ranking table plus the
derived aggregates and the rebuilt site.  Everything else changes slowly and is
handled by ``build.py``.

    python3 pyscripts/refresh_rankings.py
"""

from __future__ import annotations

import subprocess
import sys
from pathlib import Path

from wtalib import ROOT, log

STEPS = [
    "fetch_rankings.py",
    "fetch_events.py",   # completed events gain champions and results weekly
    "derive.py",
    "fetch_zh.py",       # newly ranked players need Chinese names
    "generate_data.py",
    "build_site.py",
    "verify_dashboard.py",
]


def main() -> int:
    here = Path(__file__).resolve().parent
    for script in STEPS:
        log(script, "running…")
        code = subprocess.run([sys.executable, str(here / script)], cwd=ROOT).returncode
        if code != 0:
            return code
    log("refresh", "Rankings refreshed and the site rebuilt.")
    return 0


if __name__ == "__main__":
    raise SystemExit(main())
